### PR TITLE
fix: WorkBench is now correctly spelled as Workbench 

### LINKED ISSUE #9: Correct spelling of class WorkBench
The class `WorkBench` is misspelled - it's a single word and, therefore, should be named `Workbench`.

The suggestion is to refactor the class by renaming it to the correct spelling and then creating a proxy class named `WorkBench` and marked as `Obsolete` which delegates everything to the old, renamed class. This way we avoid breaking legacy code and give our users a chance to adapt.

### LINKED ISSUE #62: WorkBench.AddToCache method won't correctly throw ArgumentException
The order of the parameters for the ArgumentException are: message, parameter name
This method, however, inverts the order

https://github.com/Avanade/Liquid-Application-Framework/blob/93f7874f2ead13d3602cc39fc1e9355206217f57/src/Liquid.Base/WorkBench.cs#L50-L56

It should also use the nameof operator

### PR DESCRIPTION
**Pull request by bruno-brant**
Changes the name of WorkBench class. I took care that this wouldn't necessarily break anything. Closes #9.

Also did a small fix on the throw instruction parameters. Closes #62.

We must first approve #141 before moving forward with this PR.

Oh, this should not have any breaking changes given that I haven't touched the sample and it's still working, but please reviewer, take a good look, since we're lacking tests.

### CODE REVIEW

File: src/Liquid.Base/WorkbenchServiceHttp.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        /// </summary>
+        GET,
+
+        /// <summary>
+        /// Represents an HTTP POST protocol method that is used to post a new entity as an addition to a URI.
+        /// </summary>
+        POST,
+
+        /// <summary>
+        /// Represents an HTTP PUT protocol method that is used to post a new entity as an addition to a URI.
```
Comment (guilhermeluizsp) [inline src/Liquid.Base/WorkbenchServiceHttp.cs]: I believe the docs for this and the other items below are incorrect. Not all values are used to post a new entity, right?

File: src/Liquid.Base/WorkbenchServiceType.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    /// </summary>
+    public enum WorkbenchServiceType
+    {
+        /// <summary>
+        /// Service used to persist objects.
+        /// </summary>
+        Repository,
+
+        /// <summary>
+        /// Service used to record events and general data in telemetry
```
Comment (guilhermeluizsp) [inline src/Liquid.Base/WorkbenchServiceType.cs]: It's a good practice to end documentation phrases with a period (.).

File: src/Liquid.Base/WorkbenchServiceType.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    [Obsolete("Prefer the correct spelled enum Liquid.Workbench")]
+    public enum WorkBenchServiceType
+    {
+        /// <summary>
+        /// Service used to persist objects.
+        /// </summary>
+        Repository = WorkbenchServiceType.Repository,
+
+        /// <summary>
+        /// Service used to record events and general data in telemetry
```
Comment (guilhermeluizsp) [inline src/Liquid.Base/WorkbenchServiceType.cs]: Same as above. It should end with a period (.).

File: test/Liquid.Base.Tests/WorkbenchTests.cs [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+        [Fact]
+        public void WhenUseLoggerThenGetLoggerReturnsSame()
+        {
+            Workbench.Instance.UseLogger<MockLogger>();
+
+            Assert.IsAssignableFrom<MockLogger>(Workbench.Instance.Logger);
+        }
+
+        [Theory, AutoSubstituteData]
+        public void GetRegisteredServiceReturnsWhatWasAddedByAddToCache(WorkbenchServiceType type, IWorkBenchService service)
```
Comment (guilhermeluizsp) [inline test/Liquid.Base.Tests/WorkbenchTests.cs]: Is there any specific reason for this test to use the misspelled `IWorkBenchService`?

File: test/Liquid.Base.Tests/WorkbenchTests.cs [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+        {
+            Workbench.Instance.AddToCache(type, service);
+
+            Assert.Same(service, Workbench.Instance.GetRegisteredService(type));
+        }
+
+        [Theory, AutoSubstituteData]
+        public void AddToCacheSameServiceTypeTwiceThrows(WorkbenchServiceType type, IWorkBenchService service1, IWorkBenchService service2)
```
Comment (guilhermeluizsp) [inline test/Liquid.Base.Tests/WorkbenchTests.cs]: Same thing as above. It should work fine with the new correctly spelled interface.

File: test/Liquid.Base.Tests/WorkbenchTests.cs [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+
+            // throws with the same service
+            Assert.ThrowsAny<Exception>(() => Workbench.Instance.AddToCache(type, service1));
+
+            // and with a different service
+            Assert.ThrowsAny<Exception>(() => Workbench.Instance.AddToCache(type, service2));
+        }
+
+        [Theory, AutoSubstituteData]
+        public void GetRegisteredServiceCallsInitializeOnService(WorkbenchServiceType type, IWorkBenchService service)
```
Comment (guilhermeluizsp) [inline test/Liquid.Base.Tests/WorkbenchTests.cs]: Again. We should move on to using the new interface to accelerate the obsolescence of the old one.

**Comment by guilhermeluizsp** [comment]
There are some breaking changes since you've renamed some interfaces (`ILightLogger` inherits from `IWorkbenchHealthCheck` now, instead of `IWorkBenchHealthCheck`, for example), but I believe they are necessary, considering the goal of this pull request.

Everything looks great, but both `ILightPolly` and `ILightWorker` still inherits from the old `IWorkBenchService` interface (the misspelled one). They don't show up in the file changes because you haven't touched them, but I think they should be using the new one.

https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/Polly/ILightPolly.cs#L10

https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/MessageBus/ILightWorker.cs#L8

**Comment by guilhermeluizsp** [comment]
@bruno-brant I'm waiting for you to merge with the master branch since #141 is approved already.

File: src/Liquid.Base/WorkbenchServiceHttp.cs (hunk 1 above)
Comment (bruno-brant) [reply src/Liquid.Base/WorkbenchServiceHttp.cs]: Yeah, actually, the text is very bad. I'll improve it.

File: src/Liquid.Base/WorkbenchServiceType.cs (hunk 2 above)
Comment (bruno-brant) [reply src/Liquid.Base/WorkbenchServiceType.cs]: Ooops

File: test/Liquid.Base.Tests/WorkbenchTests.cs (hunk 4 above)
Comment (bruno-brant) [reply test/Liquid.Base.Tests/WorkbenchTests.cs]: yes: a mistake 😁

Actually, originally, I was going to leave the tests on the old version in order to prove that old functionality still held. But I ended up moving the tests but for some reason, it seems, I stopped in the middle of it.

**Comment by bruno-brant** [comment]
> There are some breaking changes since you've renamed some interfaces (`ILightLogger` inherits from `IWorkbenchHealthCheck` now, instead of `IWorkBenchHealthCheck`, for example), but I believe they are necessary, considering the goal of this pull request.
> 
> Everything looks great, but both `ILightPolly` and `ILightWorker` still inherits from the old `IWorkBenchService` interface (the misspelled one). They don't show up in the file changes because you haven't touched them, but I think they should be using the new one.
> 
> https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/Polly/ILightPolly.cs#L10
> 
> https://github.com/Avanade/Liquid-Application-Framework/blob/939759831f202b748565a454a2828477df4d0c45/src/Liquid.Base/Interfaces/MessageBus/ILightWorker.cs#L8

Yeah, I realized that the `IWorkbenchHealthCheck` change is breaking, but I also guessed that it wouldn't be a big problem for users since, well, they shouldn't be accessing a `ILightLogger` cast to a `IWorkbenchHealthCheck`. I know that it's a possibility, but it's one I'm okay with.

Now, both polly and worker are just things that I ended up forgetting. I'll make the necessary fixes.

**Comment by bruno-brant** [comment]
Also, both Polly and Worker will be made obsolete very soon. :)

**Comment by bruno-brant** [comment]
Done, @guilhermeluizsp. The branch is rebased.

**Review by guilhermeluizsp — APPROVED** [review]

**Pull request merged**